plugins/env/app/models/environment_variable.rb

```diff
@@ -19,11 +19,12 @@ class << self
     # but return a value with a helpful message
     # also used by an external plugin
     def env(project, deploy_group, preview: false, resolve_secrets: true)
-      variables = nested_variables(project)
-      variables.sort_by!(&:priority)
-      env = variables.each_with_object({}) do |ev, all|
-        all[ev.name] = ev.value if !all[ev.name] && ev.matches_scope?(deploy_group)
-      end
+      env =
+        if (env_repo_name = ENV["DEPLOYMENT_ENV_REPO"]) && project.use_env_repo
```

Comment:
create a class method `git_repo` for `ENV["DEPLOYMENT_ENV_REPO"]` and use it for views and model, so that we can easily maintain ENV variables.